sys/net/gnrc/netif/gnrc_netif_device_type.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief   Functions that are based around a network interface's device type.
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include "log.h"
+#include "net/gnrc/netif.h"
+#include "net/ethernet.h"
+#include "net/ieee802154.h"
+
+#ifdef MODULE_GNRC_IPV6
```

Comment:
this is pretty hacky. Better only build this file in case GNRC_IPV6 is selected: put 
```
ifndef (,$(filter gnrc_ipv6,$(USEMODULE)))
  SRC += gnrc_netif_device_type.c`
endif
```
into the gnrc/netif/Makefile

Reply:
see #10546

Reply:
Nope, see #10524 where introduce stuff to this file that is not IPv6 specific, as reflected by the name.

Reply:
I could make a separate submodule `gnrc_netif_ipv6_device_type`, but that would go against the centralizing idea of #10524 (putting all device-type depending and `#ifdef` constructed functions in one place.